Fix quick terminal fullscreen toggle in front of fullscreen window

liby opened the pull request:
Fixes #7070 

The root issue occurs when the quick terminal has already entered fullscreen state and the main window is also in fullscreen mode. When attempting to toggle fullscreen again, the state detection fails due to interference from the main window's fullscreen state.

In the original code, `toggleFullscreen(mode: .nonNative)` attempts to toggle fullscreen based on the current state, but the detection is confused by the main window's fullscreen state.

liby committed:
Fix quick terminal fullscreen toggle in front of fullscreen window

Implements custom fullscreen toggle for quick terminal to avoid
presentation options conflicts when toggling fullscreen in front
of an already fullscreen window.

jparise commented on macos/Sources/Features/QuickTerminal/QuickTerminalController.swift:
```diff
@@ -485,9 +488,17 @@ class QuickTerminalController: BaseTerminalController {
 
-        // We ignore the requested mode and always use non-native for the quick terminal
-        toggleFullscreen(mode: .nonNative)
+        if savedWindowFrame != nil {
+            let originalFrame = savedWindowFrame!
```
`if let savedWindowFrame` will let you drop the force unwrap on the next line.

mitchellh commented:
Thanks! I want to understand this a bit better to see the root cause.

mitchellh reviewed (changes requested):
I think the root cause here isn't correct. 

I did some investigation and the root cause seems to be mucking around with `autoHideMenuBar` while on a space with an existing fullscreen window. I suspect it's because our accounting around our acquire/release is wrong because AppKit is injecting style masks on its own for native fullscreen.

Your fix works because it avoids all the fullscreen machinery but I'd like to keep all fullscreen logic centralized in that single place so I think it'd be more robust to detect this scenario. I'm not sure how yet but I'm looking into it...

mitchellh commented:
From logs:

```
-[NSApplication _setPresentationOptions:instance:flags:0x40e] Both NSApplicationPresentationHideMenuBar and NSApplicationPresentationAutoHideMenuBar were specified; only one is allowed
```

One of these is from macOS and one is from us. We should be able to detect this.

liby committed:
Fix macOS quick terminal fullscreen in front of fullscreen window

When a main Ghostty window is in fullscreen mode, quick terminal could get
stuck after the first fullscreen toggle due to presentation options conflicts.
This fix improves handling of menu bar visibility when toggling fullscreen
on spaces that already have fullscreen windows.

liby commented:
I just pushed a fix for the fullscreen issue. Rather than using a custom fullscreen implementation, I focused on the root cause you identified - presentation options conflict when toggling on spaces with existing fullscreen windows.

The implementation now:
- Checks for existing presentation options before applying our own
- Only applies menu hiding when not already managed by macOS
- Skips menu bar management entirely when on a space with a fullscreen window

This should avoid the conflict while keeping all the fullscreen logic in the central implementation. Thanks for pointing me in the right direction!

mitchellh commented:
Thanks, but I took a different approach in #7091. The approach you ended up taking here isn't safe because we have multiple parts of the code that could nest presentation options which is why we must not inspect or touch it directly. The issue is that macOS is modifying it causing this issue...
